src/guides/v2.3/extension-dev-guide/attributes.md

```diff
@@ -32,7 +32,7 @@ A [module](https://glossary.magento.com/module) has a set of built-in attributes
 *  visibility
 *  weight
 
-In this case, when `getCustomAttributes()` is called, the system returns only custom attributes that are not in this list.
+In this case, when `getCustomAttributes()` is called, the system returns only custom attributes that are not in this list. If you create custom attributes programmatically keep in mind to set the `system` option to false if you want to retrieve the attribute in the `custom_attributes` array.
```

Comment:
Hi @naitsirch. Thank you for your collaboration. 
Does it apply to all EAV attributes? I'm currently taking a look at [Product Model](https://github.com/magento/magento2/blob/baebe46dee141a95bf67695ee1bbf4b30af49018/app/code/Magento/Catalog/Model/Product.php#L511), which returns all attributes unless they are blacklisted in the `ProductInterface::ATTRIBUTES` constant without any additional check is the attribute system or not.

Thank you.

Reply:
Looks like you are right. But I thought that line was meant for customer attributes. After reading once again I understand that it is only referred to the catalog module.

The next sentence states 
> The Customer module does not treat its EAV attributes in a special manner. As a result, the getCustomAttributes() method returns all EAV attributes.

So this sentence must be wrong, right? Should I change this sentence and add the hint about the `system` option?

For example like this:
> The Customer module provides a `system` option for its attributes. As a result, the getCustomAttributes() method only returns those EAV attributes that are not defined as system attributes. If you create custom attributes programmatically keep in mind to set the `system` option to false if you want to retrieve the attribute in the `custom_attributes` array.

What do you think?

Reply:
That would be awesome. In that way, we will eliminate much of the confusion.

Thank you!